core/src/test/scala/pink/cozydev/protosearch/MultiIndexSuite.scala

```diff
@@ -38,6 +38,20 @@ class MultiIndexSuite extends munit.FunSuite {
     result.map(hits => hits.map(i => allBooks(i)))
   }
 
+  def searchMap(qs: String): Either[String, List[(Int, Map[String, String])]] = {
+    val q = qAnalyzer.parse(qs).leftMap(_.toString())
+    q.flatMap(mq => index.searchMap(mq.qs))
+  }
+
+  test("Term searchMap") {
+    val books = searchMap("Bad")
+    val miceMap = Map(
+      "title" -> "The Tale of Two Bad Mice",
+      "author" -> "Beatrix Potter",
+    )
+    assertEquals(books, Right(List(1 -> miceMap)))
```

Comment:
I don't want to repeat all these tests for `searchMap` we should figure out a way to reuse a suite of tests.